python/gtsam/tests/test_DsfTrackGenerator.py

```diff
@@ -0,0 +1,79 @@
+"""Unit tests for track generation using a Disjoint Set Forest data structure.
+
+Authors: John Lambert
+"""
+
+import unittest
+
+import numpy as np
+
+import gtsam
+from gtsam import (
+    DsfTrackGenerator,
+    IndexPair,
+    Keypoints,
+    KeypointsList,
+    MatchIndicesMap,
+    NamedSfmMeasurement,
+    NamedSfmMeasurementVector,
+    SfmTrack2d,
+)
+from gtsam.utils.test_case import GtsamTestCase
+
+
+class TestDsfTrackGenerator(GtsamTestCase):
+    """Tests for DsfTrackGenerator."""
+
+    def test_track_generation(self) -> None:
+        """Ensures that DSF generates three tracks from measurements in 3 images (H=200,W=400)."""
+        kps_i0 = Keypoints(coordinates=np.array([[10.0, 20], [30, 40]]))
+        kps_i1 = Keypoints(coordinates=np.array([[50.0, 60], [70, 80], [90, 100]]))
+        kps_i2 = Keypoints(coordinates=np.array([[110.0, 120], [130, 140]]))
+
+        keypoints_list = KeypointsList()
+        keypoints_list.append(kps_i0)
+        keypoints_list.append(kps_i1)
+        keypoints_list.append(kps_i2)
+
+        # For each image pair (i1,i2), we provide a (K,2) matrix of corresponding image indices (k1,k2).
+        matches_dict = MatchIndicesMap()
+        matches_dict[IndexPair(0, 1)] = np.array([[0, 0], [1, 1]])
+        matches_dict[IndexPair(1, 2)] = np.array([[2, 0], [1, 1]])
+
+        tracks = DsfTrackGenerator().generate_tracks_from_pairwise_matches(matches_dict, keypoints_list)
+        assert len(tracks) == 3
+
+        # Verify track 0.
+        assert np.allclose(tracks[0].measurements()[0].uv, np.array([10.0, 20.0]))
+        assert np.allclose(tracks[0].measurements()[1].uv, np.array([50.0, 60.0]))
+        assert tracks[0].measurements()[0].i == 0
+        assert tracks[0].measurements()[1].i == 1
+        assert tracks[0].numberMeasurements() == 2
+
+        # Verify track 1.
+        assert np.allclose(tracks[1].measurements()[0].uv, np.array([30.0, 40.0]))
+        assert np.allclose(tracks[1].measurements()[1].uv, np.array([70.0, 80.0]))
+        assert np.allclose(tracks[1].measurements()[2].uv, np.array([130.0, 140.0]))
+        assert tracks[1].measurements()[0].i == 0
+        assert tracks[1].measurements()[1].i == 1
+        assert tracks[1].measurements()[2].i == 2
+        assert tracks[1].numberMeasurements() == 3
+
+        # Verify track 2.
+        assert np.allclose(tracks[2].measurements()[0].uv, np.array([90.0, 100.0]))
+        assert np.allclose(tracks[2].measurements()[1].uv, np.array([110.0, 120.0]))
+        assert tracks[2].measurements()[0].i == 1
+        assert tracks[2].measurements()[1].i == 2
+        assert tracks[2].numberMeasurements() == 2
+
```

Comment:
I think the tests are insufficient but can be covered in a different PR. Correct me if I am wrong, but this is a naive DSF class right? So if there are tracks with multiple points in the same image, they are still considered?